tests/ui/pages/desktop/home.py

```diff
@@ -39,3 +52,120 @@ def users(self):
                 """Number of users that have downloaded the extension"""
                 users_str = self.find_element(*self._users_locator).text
                 return int(users_str.split()[0].replace(',', ''))
+
+    class FeaturedExtensions(Region):
+        """ Featured Extension region """
+
+        _root_locator = (By.ID, 'featured-extensions')
+
+        _extension_locator = (By.CSS_SELECTOR, 'section > li > div')
+        _see_all_locator = (By.CSS_SELECTOR, 'h2 > a')
+
+        @property
+        def extensions(self):
+            return [self.Extension(self.page, el) for el in self.find_elements(
+                *self._extension_locator)]
+
+        class Extension(Region):
+
+            _root_locator = (By.CLASS_NAME, 'addon-hovercard')
+
+            _name_locator = (By.CSS_SELECTOR, 'h3')
+            _link_locator = (By.TAG_NAME, 'a')
+
+            def click(self):
+                """
+                Clicks the 'See All' link within the Featured Extensions
+                region.
+                :returns: :py:class: `~pages.desktop.details.Details`
+                """
+                self.find_element(*self._link_locator).click()
+                from pages.desktop.details import Details
+                page = Details(self.selenium, self.page.base_url, timeout=30)
```

Comment:
We should avoid hard-coding timeouts. By adding this to the base page we should be able to avoid this.